kubernetes-model/src/test/java/io/fabric8/kubernetes/api/model/KubernetesRoleBindingTest.java

```diff
@@ -0,0 +1,90 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.api.model;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.rbac.KubernetesRoleBinding;
+import io.fabric8.kubernetes.api.model.rbac.KubernetesSubjectBuilder;
+import io.fabric8.kubernetes.api.model.rbac.KubernetesRoleRefBuilder;
+import io.fabric8.kubernetes.api.model.rbac.KubernetesRoleBindingBuilder;
+import org.junit.Test;
+
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_ARRAY_ORDER;
+import static net.javacrumbs.jsonunit.core.Option.IGNORING_EXTRA_FIELDS;
+import static net.javacrumbs.jsonunit.core.Option.TREATING_NULL_AS_ABSENT;
+import static net.javacrumbs.jsonunit.fluent.JsonFluentAssert.assertThatJson;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class KubernetesRoleBindingTest {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    @Test
+    public void kubernetesRoleBindingTest() throws Exception {
+        // given
+        final String originalJson = Helper.loadJson("/valid-kubernetesRoleBinding.json");
+
+        // when
+        final KubernetesRoleBinding kubernetesRoleBinding = mapper.readValue(originalJson, KubernetesRoleBinding.class);
+        final String serializedJson = mapper.writeValueAsString(kubernetesRoleBinding);
+
+        // then
+        assertThatJson(serializedJson).when(IGNORING_ARRAY_ORDER, TREATING_NULL_AS_ABSENT, IGNORING_EXTRA_FIELDS)
+                .isEqualTo(originalJson);
+    }
+
+    @Test
+    public void kubernetesRoleBuilderTest() {
+
+        KubernetesRoleBinding kubernetesRoleBinding = new KubernetesRoleBindingBuilder()
+                .withNewMetadata()
+                    .withName("read-jobs")
+                    .withNamespace("default")
+                .endMetadata()
+                .addToSubjects(0, new KubernetesSubjectBuilder()
+                        .withApiGroup("rbac.authorization.k8s.io")
+                        .withKind("User")
+                        .withName("jane")
+                        .withNamespace("default")
+                        .build()
+                )
+                .withRoleRef(new KubernetesRoleRefBuilder()
+                        .withApiGroup("rbac.authorization.k8s.io")
+                        .withKind("Role")
+                        .withName("job-reader")
+                        .build()
+                )
+                .build();
+
+        assertNotNull(kubernetesRoleBinding);
```

Comment:
Instead of doing a bunch of assertions can we serialize `mapper.writeValueAsString()` and compare with expected string.?

Reply:
Hey @hrishin 

Do you want something like 
```
// given
        final String originalJson = Helper.loadJson("/valid-kubernetesRoleBinding.json");

        // when
        final String serializedJson = mapper.writeValueAsString(kubernetesRoleBinding);

        // then
        assertThatJson(serializedJson).when(IGNORING_ARRAY_ORDER, TREATING_NULL_AS_ABSENT, IGNORING_EXTRA_FIELDS)
                .isEqualTo(originalJson);
``` 

I did this because we are asserting using mapper in another test, and in this test asserting each value.

Reply:
correct @piyush1594. This will do the same as comparing single value, right?

Reply:
But I think from asserting every value, we are doing null checks also. That's the only reason.